keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/ImageHandler.kt

```diff
@@ -53,8 +54,25 @@ class ImageHandler(
     }
   }
 
-  private fun DebianArtifact.getLatestBaseImageVersion() =
-    baseImageCache.getBaseImage(vmOptions.baseOs, vmOptions.baseLabel)
+  private suspend fun DebianArtifact.getLatestBaseImageVersion(): String {
+    val version = baseImageCache.getBaseImage(vmOptions.baseOs, vmOptions.baseLabel)
+    return findBaseAmi(version)
+  }
+
+  private suspend fun findBaseAmi(baseImage: String): String {
+    return imageService.getLatestNamedImage(baseImage, "test")
+      ?.let { namedImage ->
+        val tags = namedImage
+          .tagsByImageId
+          .values
+          .first { it?.containsKey("base_ami_version") ?: false }
```

Comment:
```suggestion
          .find { it?.containsKey("base_ami_version") ?: false }
```
...otherwise you'd get a `NoSuchElementException` if you can't find the tag, right?

Reply:
Actually, maybe rewrite the whole block like:
```kotlin
namedImage
  .tagsByImageId
  .values
  .find { it.containsKey("base_ami_version") }
  ?.map { it.get("base_ami_version") }
  ?: throw BaseAmiNotFound(baseImage)
```